nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -345,10 +352,16 @@ def setup_test_data(self, cfg):
             )
             self._test_dl = self.build_pretraining_data_loader(self._test_ds, consumed_samples)
 
-    def compute_consumed_samples(self, global_step):
+    def on_pretrain_routine_start(self) -> None:
+        # keep a copy of init_global_step
+        self.init_global_step = self.trainer.global_step
+        return super().on_pretrain_routine_start()
+
+    def compute_consumed_samples(self, global_relative_step=0):
         app_state = AppState()
         consumed_samples = (
-            global_step
+            self.init_consumed_samples
+            + global_relative_step
             * app_state.data_parallel_size
             * self.cfg.micro_batch_size
```

Comment:
Why not using here `get_num_microbatches()`?

Reply:
because it is not PP ready.